### DISCUSSION TITLE
When to handle take_persisted_messages and take_messages differently?

### DISCUSSION THREAD

**Discussion by tisonkun** [tikv/raft-rs#509] [Q&A]
It seems both methods return the messages but with different guards, `is_persisted_msg` or not.

The guard represents whether the new state is leader. In one of my scenario, the vote response can stage in message first and then this node becomes leader or not so the same message is sent in either of these two guards.

Then, I'm wondering if I should just treat these two kind of message channel the same -

1. Is there real-world usage to handle these two (logically) different messages channel with different strategies?
2. ~~Is it possible to just export `is_persisted_msg` then let the caller decide whether to take messages?~~
    * UPDATE: Since I know where the messages come from, this can be done now.

cc @BusyJay

**Comment by BusyJay** [comment]
I think it's explained in the docs:

> Persisted Messages specifies outbound messages to be sent AFTER the HardState, Entries and Snapshot are persisted to stable storage.

It's OK to treat messages as persisted message, though it may increase latency as follower may take more time to learn new logs.
